ouroboros-network/src/Ouroboros/Network/Block.hs

```diff
@@ -139,12 +157,14 @@ blockPoint b =
 --
 data ChainUpdate block = AddBlock block
                        | RollBack (Point block)
-  deriving (Eq, Show)
+
+deriving instance (Show block, Show (HeaderHash block)) => Show (ChainUpdate block)
+deriving instance (Eq block, Eq (HeaderHash block)) => Eq (ChainUpdate block)
 
 mapChainUpdate :: HeaderHash a ~ HeaderHash b
                => (a -> b) -> ChainUpdate a -> ChainUpdate b
 mapChainUpdate f (AddBlock b) = AddBlock $ f b
-mapChainUpdate _ (RollBack p) = RollBack $ castPoint p
+mapChainUpdate _ (RollBack p) = RollBack $ p
```

Comment:
the  `$` is redundant here, though I guess you left if to align things, so it's fine.